src/image.c

```diff
@@ -472,6 +472,104 @@ static void wolfBoot_verify_signature(uint8_t key_slot,
 }
 #endif /* WOLFBOOT_SIGN_XMSS */
 
+#ifdef WOLFBOOT_SIGN_ML_DSA
+#include <wolfssl/wolfcrypt/dilithium.h>
+static void wolfBoot_verify_signature(uint8_t key_slot,
+        struct wolfBoot_image *img, uint8_t *sig)
+{
+    int       ret = 0;
+    MlDsaKey  ml_dsa;
+    uint8_t * pubkey = NULL;
+    int       pub_len = 0;
+    int       sig_len = 0;
+    int       verify_res = 0;
+
+    wolfBoot_printf("info: ML-DSA wolfBoot_verify_signature\n");
+
+    pubkey = keystore_get_buffer(key_slot);
+
+    if (pubkey == NULL) {
+        wolfBoot_printf("error: ML-DSA pubkey not found\n");
+        return;
+    }
+
+    ret = wc_MlDsaKey_Init(&ml_dsa, NULL, INVALID_DEVID);
+
+    if (ret != 0) {
+        wolfBoot_printf("error: wc_MlDsaKey_Init returned %d\n", ret);
+    }
+
+    if (ret == 0) {
+        /* Set the ML-DSA security level. */
+        ret = wc_MlDsaKey_SetParams(&ml_dsa, ML_DSA_LEVEL);
+
+        if (ret != 0) {
+            wolfBoot_printf("error: wc_MlDsaKey_SetParams(%d)" \
+                            " returned %d\n", ML_DSA_LEVEL, ret);
+        }
+    }
+
+    /* Make sure pub key matches parameters. */
+    if (ret == 0) {
+        ret = wc_MlDsaKey_GetPubLen(&ml_dsa, &pub_len);
+
+        if (ret != 0 || pub_len <= 0) {
+            wolfBoot_printf("error: wc_MlDsaKey_GetPubLen returned %d\n", ret);
+        }
+        else if (pub_len != KEYSTORE_PUBKEY_SIZE_ML_DSA) {
+            wolfBoot_printf("error: ML-DSA pub key mismatch: got %d bytes " \
+                   "expected %d\n", pub_len, KEYSTORE_PUBKEY_SIZE_ML_DSA);
+            ret = -1;
+        }
+    }
+
+    /* Make sure sig len matches parameters. */
+    if (ret == 0) {
+        ret = wc_MlDsaKey_GetSigLen(&ml_dsa, &sig_len);
+
+        if (ret != 0 || sig_len <= 0) {
+            wolfBoot_printf("error: wc_MlDsaKey_GetPubLen returned %d\n", ret);
+        }
+        else if (sig_len != IMAGE_SIGNATURE_SIZE) {
+            wolfBoot_printf("error: ML-DSA sig len mismatch: got %d bytes " \
+                   "expected %d\n", sig_len, IMAGE_SIGNATURE_SIZE);
+            ret = -1;
+        }
+    }
+
+    if (ret == 0) {
+        /* Now import pub key. */
+        ret = wc_MlDsaKey_ImportPubRaw(&ml_dsa, pubkey, pub_len);
```

Comment:
```
davidgarske@Davids-MacBook-Pro-2 wolfBoot-alt % make
        [CC-sim] src/string.o
        [CC-sim] src/image.o
src/image.c:542:15: error: use of undeclared identifier 'out'
  542 |         ret = wc_MlDsaKey_ImportPubRaw(&ml_dsa, pubkey, pub_len);
      |               ^
lib/wolfssl/wolfssl/wolfcrypt/dilithium.h:753:32: note: expanded from macro 'wc_MlDsaKey_ImportPubRaw'
  753 |     wc_dilithium_import_public(out, outLen, key)
      |                                ^
src/image.c:542:15: error: use of undeclared identifier 'outLen'
lib/wolfssl/wolfssl/wolfcrypt/dilithium.h:753:37: note: expanded from macro 'wc_MlDsaKey_ImportPubRaw'
  753 |     wc_dilithium_import_public(out, outLen, key)
      |                                     ^
2 errors generated.
make: *** [src/image.o] Error 1
```

Reply:
```
cp config/examples/sim-ml-dsa.config .config
make distclean
make
```

Reply:
Have to build with this branch:
- https://github.com/wolfSSL/wolfssl/pull/7983